classes/PrestaShopAutoload.php

```diff
@@ -124,37 +134,34 @@ public function load($className)
         }
 
         // regenerate the class index if the requested file doesn't exists
-        if ((isset($this->index[$className]) && $this->index[$className]['path'] && !is_file($this->root_dir . $this->index[$className]['path']))
-            || (isset($this->index[$className . 'Core']) && $this->index[$className . 'Core']['path'] && !is_file($this->root_dir . $this->index[$className . 'Core']['path']))
+        if ((isset($this->index[$className]) && $this->index[$className]['path'] && !is_file($this->root_dir . '/' . $this->index[$className]['path']))
+            || (isset($this->index[$className . 'Core']) && $this->index[$className . 'Core']['path'] && !is_file($this->root_dir . '/' . $this->index[$className . 'Core']['path']))
```

Comment:
Seeing all the direct usages of `$this->root_dir`, I am thinking it would be interesting to add 
```php
private function getAbsolutePath(string relativePath): string|false {
    return realpath($this->root_dir . '/' . relativePath);
}
```
This way, if you change anything about the way you build bath, it would be easy to update :thinking: 

What do you think ?